src/utils.ts

```diff
@@ -62,6 +62,28 @@ export function max(a: BigNumberish, b: BigNumberish): BN {
 
 export const fixedPointAdjustment = toBNWei("1");
 
+/**
+ * Takes two values and returns a list of number intervals
+ *
+ * @example
+ * ```js
+ * getSamplesBetween(1, 9, 3) //returns [[1, 3], [4, 7], [8, 9]]
+ * ```
+ */
```

Comment:
Can you give more context here?

1. These aren't arbitrary intervals in that range? How are they picked?
2. What is this meant to do? Is there an example of how this might be useful?

Reply:
This just splits an interval into multiple intervals that have the same length. Usage? To get the block ranges between blockNumber1 and blockNumber2 with each range having a max length of x (the third parameter of the function)